### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by Philonous**
I had implemented (a less complete) version of AuthnRequests in another project and was about to work on merging it here, so I'm pleased to see that the work is already done :sweat_smile: 

this PR adds the references/notes I used took my implementation, since I found that they kept coming in handy. I hope they can be useful here as well.

* I have also added a mention of `Network.Wai.SAML2.Request` in the changelog and updated the README

### CODE REVIEW

**Comment by Philonous** [comment]
I folded a few more documentation changes into this PR and changed the title to reflect this.

File: README.md [hunk 1]
```diff
@@ -5,17 +5,16 @@
...
+A Haskell library which implements SAML2 assertion validation as WAI middleware. This can be used by a Haskell web application (the service provider, SP) to perform identity provider (IdP) initiated authentication, i.e. SAML2-based authentication where the authentication begins at the IdP-end, the IdP authenticates the user, and then gets the user to submit a SAML2 assertion back to the SP (known as "unsolicited SSO" within e.g. [the Shibboleth project](https://wiki.shibboleth.net/confluence/display/IDP30/UnsolicitedSSOConfiguration#UnsolicitedSSOConfiguration-SAML2.0)).
 
 ## Completeness
 
 There are currently a number of limitations to this library:
 
-* As mentioned above, while the library implements IdP-initiated authentication, it does not yet implement SP-initiated authentication (where the SP submits a login request to the IdP).
+* The library implements IdP-initiated authentication and has some support for SP-initiated authentication (See Network.Wai.SAML2.Request)
```
Comment (mbg) [inline README.md]:
```suggestion
* The library implements IdP-initiated authentication and has some support for SP-initiated authentication (See the documentation for the `Network.Wai.SAML2.Request` module)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * Exported `NameID` (formerly `NameId`), and renamed `subjectNameId` to `subjectNameID`
+* Add new module `Network.Wai.SAML2.Request` with AuthnRequest generation for SP-initiated login flow
```
Comment (mbg) [inline CHANGELOG.md]:
```suggestion
* Add new module `Network.Wai.SAML2.Request` with `AuthnRequest` generation for SP-initiated login flow ([#19](https://github.com/mbg/wai-saml2/pull/19) by [@fumieval](https://github.com/fumieval))
```

**Review by mbg — APPROVED** [review]
This is awesome, thank you for contributing this @Philonous!

I am sorry for not reviewing this and your other PRs + issues for longer than I'd like. Unfortunately, I didn't have much time in the last few weeks, but I should be able to work through them now, hopefully!

I suggested a couple of minor improvements and this branch will probably require a rebase before I can merge it.

**Comment by Philonous** [comment]
I added the suggestions and rebased the branch

**Comment by mbg** [comment]
Thank you! 🚀 And sorry again for the delay in reviewing this.

**Pull request merged**